pockemun/src/api/getPokemonByTypesName.ts

```diff
@@ -0,0 +1,13 @@
+import axios from 'axios'
+
+import type { PokemonsByType } from '@/api/types'
+
+const getPokemonsByTypeName = async (name: string) => {
+  const baseUrl = 'https://pokeapi.co/api/v2/type/'
+  const url = `${baseUrl}/${name}/`
+
+  const response = await axios.get<PokemonsByType>(url)
```

Comment:
There is nothing wrong with `axios` but js has an inbuild `fetch` method which is just as simple to use and good to know. So just as a practice you could try to re-write this using fetch:
https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch